sdk/test/trace/batch_span_processor_test.cc

```diff
@@ -0,0 +1,344 @@
+
+    auto batch_processor = GetMockProcessor(spans_received, is_shutdown);
+    const int num_spans = 3;
+
+    auto test_spans = GetTestSpans(batch_processor, num_spans);
+
+    for(int i = 0; i < num_spans; ++i){
+        batch_processor->OnEnd(std::move(test_spans->at(i)));
+    }
+
+    // Wait a bit to add spans to the buffer and for the background worker
+    // thread of the batch processor to wait on a condition variable.
+    // Helps in proper preemption of the worker thread.
+    // NOTE: If this delay is not added, it's possible for Shutdown() to notify the
+    //       worker thread even before it starts waiting, causing it to sleep for
+    //       the entire 'schedule_delay_millis' timeout
+    // TODO: Discuss this
+    std::this_thread::sleep_for(std::chrono::milliseconds(10));
+
+    batch_processor->Shutdown();
+
+    EXPECT_EQ(num_spans, spans_received->size());
+    for(int i = 0; i < num_spans; ++i)
+    {
+        EXPECT_EQ("Span " + i, spans_received->at(i)->GetName());
+    }
+   
+    EXPECT_TRUE(*is_shutdown);
+}
+
+TEST_F(BatchSpanProcessorTestPeer, TestForceFlush)
+{
+    std::shared_ptr<bool> is_shutdown(new bool(false));
+    std::shared_ptr<std::vector<std::unique_ptr<sdk::trace::SpanData>>> spans_received(
+        new std::vector<std::unique_ptr<sdk::trace::SpanData>>);
+
+    auto batch_processor = GetMockProcessor(spans_received, is_shutdown);
+    const int num_spans = 3;
+
+    auto test_spans = GetTestSpans(batch_processor, num_spans);
+
+    for(int i = 0; i < num_spans; ++i){
+        batch_processor->OnEnd(std::move(test_spans->at(i)));
+    }
+
+    // Wait a bit here too for the same reasons as mentioned in the previous test (TestShutdown)
+    // Helps in proper preemption of the worker thread
+    std::this_thread::sleep_for(std::chrono::milliseconds(10));
+
+    // force flush
+    batch_processor->ForceFlush(std::chrono::milliseconds(100));
+
+    EXPECT_EQ(num_spans, spans_received->size());
+    for(int i = 0; i < num_spans; ++i)
+    {
+        EXPECT_EQ("Span " + i, spans_received->at(i)->GetName());
+    } 
+
+    // Create some more spans to make sure that the processor still works
+    auto more_test_spans = GetTestSpans(batch_processor, num_spans);
+    for(int i = 0; i < num_spans; ++i){
+        batch_processor->OnEnd(std::move(more_test_spans->at(i)));
+    }
+
+    std::this_thread::sleep_for(std::chrono::milliseconds(10));
+    batch_processor->ForceFlush(std::chrono::milliseconds(100));
+
+    EXPECT_EQ(num_spans*2, spans_received->size());
+    for(int i = 0; i < num_spans; ++i)
+    {
+        EXPECT_EQ("Span " + i%3, spans_received->at(i)->GetName());
+    } 
+    // Wait a bit here so that the worker thread can go back to waiting on condition
+    // variable and the shutdown() method can subsequently wake the worker thread
+    // TODO: Discuss this
+    std::this_thread::sleep_for(std::chrono::milliseconds(10));
```

Comment:
Please have a look at this too. It's for the same reason as the [previous comment](https://github.com/open-telemetry/opentelemetry-cpp/pull/195/files#r457811438).